### PR TITLE
[ELY-1987] Add the ability to adjust the case of a user name using an Elytron principal transformer

### PR DESCRIPTION
**Pull request by SoniaZaldana**
https://issues.redhat.com/browse/ELY-1987

### CODE REVIEW

File: auth/util/src/main/java/org/wildfly/security/auth/util/CaseNameRewriter.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+     * Construct a new instance which transforms to upper case.
+     */
+    public CaseNameRewriter() {
+        this.upperCase = true;
+    }
+
+    /**
+     * Construct a new instance.
+     *
+     * @param upperCase - converts to upper case if set to true. Converts to
```
Comment (fjuma) [inline auth/util/src/main/java/org/wildfly/security/auth/util/CaseNameRewriter.java:42]: Just a very minor style comment - we usually don't include a dash here. We also wrap `true` and `false` with the {@code} tag, e.g. something like,

`upperCase {@code true} if the principal should be converted to upper case, {@code false} if the principal should be converted to lower case`

File: auth/util/src/main/java/org/wildfly/security/auth/util/CaseNameRewriter.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * @author <a href="mailto:szaldana@redhat.com">Sonia Zaldana Calles</a>
+ */
+public final class CaseNameRewriter implements NameRewriter {
+    private final boolean upperCase;
+
+    /**
+     * Construct a new instance which transforms to upper case.
+     */
+    public CaseNameRewriter() {
+        this.upperCase = true;
```
Comment (fjuma) [inline auth/util/src/main/java/org/wildfly/security/auth/util/CaseNameRewriter.java:36]: The one-argument constructor can just be called here instead:

`this(true)`

File: tests/base/src/test/java/org/wildfly/security/auth/server/CaseNameRewriterTest.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
+package org.wildfly.security.auth.server;
```
Comment (fjuma) [inline tests/base/src/test/java/org/wildfly/security/auth/server/CaseNameRewriterTest.java:1]: Missing copyright header

File: auth/util/src/main/java/org/wildfly/security/auth/util/CaseNameRewriter.java (hunk 1 above)
Comment (SoniaZaldana) [reply auth/util/src/main/java/org/wildfly/security/auth/util/CaseNameRewriter.java]: fixed!

File: auth/util/src/main/java/org/wildfly/security/auth/util/CaseNameRewriter.java (hunk 2 above)
Comment (SoniaZaldana) [reply auth/util/src/main/java/org/wildfly/security/auth/util/CaseNameRewriter.java]: fixed!

File: tests/base/src/test/java/org/wildfly/security/auth/server/CaseNameRewriterTest.java (hunk 3 above)
Comment (SoniaZaldana) [reply tests/base/src/test/java/org/wildfly/security/auth/server/CaseNameRewriterTest.java:18]: fixed!

**Comment by fjuma** [comment]
@SoniaZaldana This PR has a conflict to be resolved whenever you get a chance.

**Comment by SoniaZaldana** [comment]
@fjuma I resolved the conflicts.

**Pull request merged**